src/core/theme/index.ts

```diff
@@ -1,26 +1,73 @@
-import { typography } from './typography';
+import { FlattenSimpleInterpolation } from 'styled-components';
+import { typographyTokens } from './typography';
+import { typographyUtils } from './utils';
 import { colors, shadows, gradients, outlines } from './colors';
 import { spacing } from './spacing';
 import { zindexes } from './zindexes';
 import { transitions } from './transitions';
 import { radius } from './radius';
 
-export type ThemeProp = typeof suomifiTheme;
-export type ColorProp = keyof typeof suomifiTheme.colors;
+export type TokensProp = typeof importedTokens;
+export type DefaultTokensProp = typeof defaultTokens;
+export type TypographyProp = keyof typeof importedTokens.typography;
+export type ColorProp = keyof typeof importedTokens.colors;
+export type SpacingProp = keyof typeof importedTokens.spacing;
+export type SuomifiThemeProp = ReturnType<typeof suomifiTheme>;
 
-export interface ThemeComponent {
-  /** Default as suomifiTheme */
-  theme?: ThemeProp;
+export interface TokensComponent {
+  tokens?: TokensProp;
 }
 
-export const suomifiTheme = {
-  typography,
-  colors,
+export interface SuomifiThemeComponent {
```

Comment:
Why is this named as a component? Feels confusing for me as this is only a class with properties?